### PR TITLE
Clean up/use framework in DSUsageBitsErrors

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Cleaned up DSUsageBitsError, using the framework calls to reduce boilerplate code to improve maintainability and readability.  Added helper calls as needed.  Addresses (fixes?) #2184

### CODE REVIEW

File: tests/vktestbinding.h [hunk 1]
```diff
@@ -615,13 +630,30 @@ class DescriptorPool : public internal::NonDispHandle<VkDescriptorPool> {
     std::vector<DescriptorSet *> alloc_sets(const Device &dev, const DescriptorSetLayout &layout, uint32_t count);
     DescriptorSet *alloc_sets(const Device &dev, const DescriptorSetLayout &layout);
 
+    template <typename PoolSizes>
```
Comment (jzulauf-lunarg) [inline tests/vktestbinding.h]: The interface is written to be agnostic to vector or array (both of which support data and size), but if folks find it too odd, I can wind it back to just vector.

**Comment by jzulauf-lunarg** [comment]
@chrisforbes I wasn't clear on the "break apart" element of #2184.  It's just a single loop over all the issues, once you clean the cruft.  The clean up saves about half the lines, but there's any number of other tests that could shed line count with a similar application of helpers.

**Comment by jzulauf-lunarg** [comment]
Possible to get a quick review to merge and close?

**Comment by chrisforbes** [comment]
@jzulauf-lunarg the 'break apart' was to make failures easy to interpret from just the test name. These complex tests with many subtests always require more digging.

**Pull request merged**